### PR TITLE
98360 landing page renders different alerts for user actionable errors and all other errors

### PR DESCRIPTION
**Pull request by kjsuarez**
## Are you removing, renaming or moving a folder in this PR?
- [x] No, I'm not changing any folders (skip to TeamSites and delete the rest of this section)
- [ ] Yes, I'm removing, renaming or moving a folder

### :warning: TeamSites :warning:
Examples of a TeamSite: https://va.gov/health and https://benefits.va.gov/benefits/. This scenario is also referred to as the "injected" header and footer. You can reach out in the `#sitewide-public-websites` Slack channel for questions.

Did you change site-wide styles, platform utilities or other infrastructure?
- [x] No
- [ ] Yes, and I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#that-sounds-normal-so-whats-the-proxy-all-about) to test the injected header scenario

## Summary
Handle general errors when hitting the mhv account creation api.

## Related issue(s)
[98360](https://github.com/department-of-veterans-affairs/va.gov-team/issues/98360)

## Testing done
Tests added

## Screenshots
Screenshot 2024-12-19 at 2 51 53 PM
Screenshot 2024-12-19 at 2 52 48 PM

## What areas of the site does it impact?
MHV landing page

## Acceptance criteria
When API returns any error code _not_ in this list: [801,805,806,807] a specific alert should be rendered and certain cards should change.

### Quality Assurance & Testing
- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [x] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/prepare-for-an-accessibility-staging-review) has been performed

### Authentication

- [x] Did you login to a local build and verify all authenticated routes work as expected with a test user

### CODE REVIEW

File: src/applications/mhv-landing-page/selectors/mhvAccountStatus.js [hunk 1]
```diff
@@ -11,6 +11,17 @@ export const mhvAccountStatusUsersuccess = state => {
...
+export const mhvAccountStatusErrorsSorted = state => {
+  if (state?.myHealth?.accountStatus?.data?.errors) {
+    return state?.myHealth?.accountStatus?.data?.errors.sort(
+      (a, b) =>
+        userActionErrorCodes.includes(b.code) -
+        userActionErrorCodes.includes(a.code),
+    );
+  }
+  return [];
+};
```
Comment (nichia) [inline src/applications/mhv-landing-page/selectors/mhvAccountStatus.js:23]: Can you talk more about what you want to do here?
From manual local testing with this mock data (assuming that we can get multiple error codes):
```
[{"title":"The server responded with status 422","detail":"things fall apart","code":"802"},{"title":"The server responded with status 500","detail":"things fall apart","code":"500"},{"title":"The server responded with status 422","detail":"things fall apart","code":"805"},{"title":"The server responded with status 422","detail":"things fall apart","code":"801"}]
```
The output of mhvAccountStatusErrorsSorted is:
```
[{"title":"The server responded with status 422","detail":"things fall apart","code":"805"},{"title":"The server responded with status 422","detail":"things fall apart","code":"801"},{"title":"The server responded with status 422","detail":"things fall apart","code":"802"},{"title":"The server responded with status 500","detail":"things fall apart","code":"500"}]
```
Is this the expected output?

Comment (nichia) [reply src/applications/mhv-landing-page/selectors/mhvAccountStatus.js:23]: Actually, maybe we cannot get a mix of error status 422 & 500?

Comment (kjsuarez) [reply src/applications/mhv-landing-page/selectors/mhvAccountStatus.js:23]: The idea is to prioritize user actionable errors. If the api somehow returns multiple errors, some of which are user actionable and some of which are not, I consider the user errors to be the most important thing to display. So I sort the errors based on if they are in the user-actionable list and I display the top of the array.

Comment (dcloud) [reply src/applications/mhv-landing-page/selectors/mhvAccountStatus.js:23]: Copying in my slack comment:

> Maybe in the refactor we can rename the variable and/or function to indicate the kind of sorting/prioritization (or add a code comment)

There may be a few ways to clarify the purpose of this code for our future selves.

Something for the next iteration, perhaps?

Comment (dcloud) [reply src/applications/mhv-landing-page/selectors/mhvAccountStatus.js:23]: `mhvAccountErrorsActionablePriority` is the closest I can get right now, and that's clunky

**Review by dcloud — APPROVED** [review]
Thanks for the code comment. This looks good to me!

File: src/applications/mhv-landing-page/selectors/mhvAccountStatus.js (hunk 1 above)
Comment (nichia) [reply src/applications/mhv-landing-page/selectors/mhvAccountStatus.js:25]: Based on the above output of mhvAccountStatusErrorsSorted mock data, the rendered error code for alert & card need to be updated to use the same data.
Screenshot 2024-12-19 at 3 48 35 PM

File: src/applications/mhv-landing-page/components/alerts/AlertMhvNoAction.jsx [hunk 2]
```diff
@@ -0,0 +1,72 @@
+import React, { useEffect } from 'react';
+import PropTypes from 'prop-types';
+// eslint-disable-next-line import/no-named-default
+import { default as recordEventFn } from '~/platform/monitoring/record-event';
+import { VaAlert } from '@department-of-veterans-affairs/component-library/dist/react-bindings';
+
+const AlertMhvNoAction = ({ errorCode, testId, recordEvent }) => {
+  const headline = `You can't access messages, medications, or medical records right now`;
```
Comment (nichia) [inline src/applications/mhv-landing-page/components/alerts/AlertMhvNoAction.jsx:8]: We have the `title` prop for this component. Do we want to use that for headline?
 `title: "You can't access messages, medications, or medical records right now",`

**Pull request merged**